core/dictgen/src/rootcling_impl.cxx

```diff
@@ -4700,9 +4700,20 @@ int RootClingMain(int argc,
 
    int rootclingRetCode(0);
 
-   if (linkdef.empty()) {
-      // There is no linkdef file, we added the 'default' #pragma to
-      // interpPragmaSource.
+   if (linkdef.empty() || ROOT::TMetaUtils::IsLinkdefFile(linkdefFilename.c_str())) {
+      if (ROOT::TMetaUtils::IsLinkdefFile(linkdefFilename.c_str())) {
```

Comment:
I don't know what `linkdefFilename` looks like for `linkdef.empy()`?
```suggestion
      if (!linkdef.empty() && ROOT::TMetaUtils::IsLinkdefFile(linkdefFilename.c_str())) {
```

Reply:
These are two different cases. If linkdef is empty we synthesize a default content. I think this change is not what we want.